### PR TITLE
Fix for async mode in wdio-sync

### PR DESCRIPTION
**Pull request by SanthoshBonala**
## Proposed changes

Fix for the [Issue ](https://github.com/webdriverio/webdriverio/issues/4008)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/wdio-sync/src/index.js [hunk 1]
```diff
@@ -118,7 +118,7 @@ const runHook = function (hookFn, origFn, before, after, repeatTest = 0) {
             /**
              * user wants handle async command using promises, no need to wrap in fiber context
              */
-            if (hookFn.name === 'async') {
+            if (hookFn.constructor.name === 'AsyncFunction') {
                 return executeAsync.call(this, hookFn, repeatTest, filterSpecArgs(hookArgs))
```
Comment (mgrybyk) [inline packages/wdio-sync/src/index.js]: Is it possible that hookFn can be undefined or something without constructor property? 
In such case we'll get cannot read property name of undefined?
Do you think it makes sense to verify if constructor property exists in hookFn?

Comment (SanthoshBonala) [reply packages/wdio-sync/src/index.js]: sure updating the pr to check for constructor property on function prototype

Comment (mgrybyk) [reply packages/wdio-sync/src/index.js]: Do you think we still need to keep the old behavior? 
Is it possible at all to name function async? Isn't it a keyword? (Can't verify right now, just asking)

Comment (SanthoshBonala) [reply packages/wdio-sync/src/index.js]: it is possible to name fn as aync eventhough async is a keyword. Having an old workflow can be used as a fallback approach? Do you want me to remove old behavior?

Comment (mgrybyk) [reply packages/wdio-sync/src/index.js]: Not sure if anyone used it, we can keep it, I don't mind. Let's see if anyone else has some input of this.

I still think we can remove this one `typeof hookFn === 'function' `

Comment (SanthoshBonala) [reply packages/wdio-sync/src/index.js]: sure updating

Comment (mgrybyk) [reply packages/wdio-sync/src/index.js]: Thanks a lot!

Comment (SanthoshBonala) [reply packages/wdio-sync/src/index.js]: no problem

**Pull request merged**